lib/dialogs/add-state.js

```diff
@@ -0,0 +1,4 @@
+module.exports = (dialog, channel, resource) => {
```

Comment:
Can we add a comment here explaining what this function is for? Something like:

> This function adds a `state` property for dialogs, in order to carry additional, hidden, information to the webhook received when the dialog is submitted. Typically this information will be used in middleware functions and injected into attributes of `res.locals`.

Reply:
Good call!